edx_exams/apps/router/middleware.py

```diff
@@ -0,0 +1,32 @@
+"""
+Middleware that checks if in incoming request should be
+routed to the legacy proctoring service (edx-proctoring)
+"""
+from django.utils.deprecation import MiddlewareMixin
+
+from edx_exams.apps.api.v1.views import CourseExamsView
+from edx_exams.apps.core.models import CourseExamConfiguration
+from edx_exams.apps.router.views import CourseExamsOverrideView
+
+OVERRIDE_VIEW_MAP = {
+    CourseExamsView: CourseExamsOverrideView
+}
+
+class ExamRequestMiddleware(MiddlewareMixin):
+    """
+    Intercept requests and determine if exams for this course
+    should be managed by legacy edx-proctoring system. If so, requests
+    are intercepted and forwarded to that service
+    """
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        try:
+            override_view = OVERRIDE_VIEW_MAP.get(view_func.view_class)
+        except AttributeError:
+            override_view = None
```

Comment:
this feels like it would make more sense in form of an if
if func in map
view = view
else view = none

because it's not really an error, it's just not in the map

Reply:
the reason I have this here is some functions passed to this middleware may not have a view_class attribute so this simplifies having to check if that attribute exists first.